stmhal/usbd_conf.c

```diff
@@ -1,587 +1,654 @@
+  *
+  * Unless required by applicable law or agreed to in writing, software 
+  * distributed under the License is distributed on an "AS IS" BASIS, 
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  *
+  ******************************************************************************
+  */
+
+/* Includes ------------------------------------------------------------------*/
+#include STM32_HAL_H
+#include "usbd_core.h"
+#include "py/obj.h"
+#include "irq.h"
+
+/* Private typedef -----------------------------------------------------------*/
+/* Private define ------------------------------------------------------------*/
+/* Private macro -------------------------------------------------------------*/
+/* Private variables ---------------------------------------------------------*/
+PCD_HandleTypeDef pcd_handle;
+
+/* Private function prototypes -----------------------------------------------*/
+/* Private functions ---------------------------------------------------------*/
+
+/*******************************************************************************
+                       PCD BSP Routines
+*******************************************************************************/
+/**
+  * @brief  Initializes the PCD MSP.
+  * @param  hpcd: PCD handle
+  * @retval None
+  */
+void HAL_PCD_MspInit(PCD_HandleTypeDef *hpcd)
+{
+  GPIO_InitTypeDef  GPIO_InitStruct;
+  
+  if(hpcd->Instance == USB_OTG_FS)
+  {
+    /* Configure USB FS GPIOs */
+    __GPIOA_CLK_ENABLE();
+
+    GPIO_InitStruct.Pin = (GPIO_PIN_11 | GPIO_PIN_12);
+    GPIO_InitStruct.Mode = GPIO_MODE_AF_PP;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    GPIO_InitStruct.Speed = GPIO_SPEED_HIGH;
+    GPIO_InitStruct.Alternate = GPIO_AF10_OTG_FS;
+    HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+    /* Configure VBUS Pin */
+#if defined(MICROPY_HW_USB_VBUS_DETECT_PIN)
+    // USB VBUS detect pin is always A9
+    GPIO_InitStruct.Pin = GPIO_PIN_9;
+    GPIO_InitStruct.Mode = GPIO_MODE_INPUT;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+#endif
+
+#if defined(MICROPY_HW_USB_OTG_ID_PIN)
+    // USB ID pin is always A10
+    GPIO_InitStruct.Pin = GPIO_PIN_10;
+    GPIO_InitStruct.Mode = GPIO_MODE_AF_OD;
+    GPIO_InitStruct.Pull = GPIO_PULLUP;
+    GPIO_InitStruct.Alternate = GPIO_AF10_OTG_FS;
+    HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+#endif
+
+    /* Enable USB FS Clocks */ 
+    __USB_OTG_FS_CLK_ENABLE();
+    
+    /* Set USBFS Interrupt priority */
+    HAL_NVIC_SetPriority(OTG_FS_IRQn,  IRQ_PRI_OTG_FS, IRQ_SUBPRI_OTG_FS);
+    
+    /* Enable USBFS Interrupt */
+    HAL_NVIC_EnableIRQ(OTG_FS_IRQn);
+  } 
+  
+#if defined(USE_USB_HS)
+  else if(hpcd->Instance == USB_OTG_HS)
+  {
+    
+#if defined(STM32F429DISC)
+    /* Configure USB FS GPIOs */
+    __GPIOB_CLK_ENABLE();
```

Comment:
I will define USE_USB_HS_IN_FS in stm32xx_hal_conf.h and change usbd_conf.c accordingly. I remove checks for the board (STM32F4DISC define) in usbd_conf.c, too. Then STM32F4DISC is no more needed and I remove that board define. If needed it can be added latter again.